galaxy/importer/linters/__init__.py

```diff
@@ -68,16 +68,26 @@ def parse_id_and_desc(self, message):
             rule_desc = ' '.join(msg_parts[2:])
 
             error_id = msg_parts[1]
+            # REVIEW(cutwater): Flake8 is configured to return `E,F,W` rules.
+            #   Why is parsing rules prefixes list differs
+            #   from FLAKE8_SELECT_ERRORS.
             if error_id[0] not in ['E', 'W']:
                 error_id = None
 
         except IndexError:
+            # REVIEW(cutwater): If this except clause is executed,
+            #  rule_desc may not be defined, but it's used below.
             error_id = None
```

Comment:
If this except clause is executed, error_id should be None and the next code block should return before rule_desc is referenced

Reply:
Indeed, however the code flow is fuzzy and it's recognized by static code analysis tools as a potential issue.
You could've invert the logic and return error_id and message if it's found and handle error case down below.

```
        try:
            error_id, rule_desc = message.split(' ', 1)
        except ValueError:
            pass
        else:
            if error_id in ['E', 'W']:
                return error_id, rule_desc

        # exception is probably better here
        logger.warning('Cannot parse linter message: {}'.format(message))
        return None, None
```